feat: Ismp

Ad96el opened the pull request:
# Implements ISMP related pallet

Ad96el committed:
intermediate

Ad96el committed:
compiling ismp

Ad96el committed:
ismp

Ad96el committed:
working no std leak

Ad96el committed:
compiling

Ad96el committed:
config

Ad96el committed:
compiling

Ad96el committed:
compiling

Ad96el committed:
make ci pass

Ad96el committed:
benchmarking

Ad96el committed:
Merge branch 'chore/polkadot-1.8.0' into ismp

Ad96el committed:
lock

Ad96el committed:
remove unused import

Ad96el committed:
clean up

Ad96el committed:
clean up

abdulmth reviewed (commented):
Looks good to me

Wizdave97 commented on nodes/parachain/src/service.rs:
```diff
@@ -270,11 +271,13 @@ where
 			let deps = crate::rpc::FullDeps {
 				client: Arc::clone(&client),
 				pool: Arc::clone(&transaction_pool),
+				backend: Arc::clone(&backend),
```
You need to add `ismp-parachain-inherent` to the`create_inherent_providers` in `AuraParams`
guide:  https://docs.hyperbridge.network/developers/polkadot/ismp-parachain-inherent

Ad96el replied:
[db428f9](https://github.com/KILTprotocol/kilt-node/pull/888/commits/db428f9be256e3faef852730780bbe6a12408c0f)

Wizdave97 commented on runtimes/peregrine/src/ismp.rs:
```diff
@@ -0,0 +1,103 @@
+
+impl pallet_token_gateway::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Dispatcher = Ismp;
```
Dispatcher should be pallet-hyperbridge
```suggestion
	type Dispatcher = Hyperbridge;
```

Ad96el replied:
[e5a21b4](https://github.com/KILTprotocol/kilt-node/pull/888/commits/e5a21b4f2045c2dbf1c14f3e8bf8823f68ed6f6e)

Wizdave97 reviewed (commented):
Signed Extras in your runtime must be an exact match of what is described in the link below
https://docs.hyperbridge.network/developers/polkadot/pallet-ismp#signed-extensions

Enable offchain indexing in your node by default, follow the guide below:
https://docs.hyperbridge.network/developers/polkadot/pallet-ismp#offchain-indexing

Wizdave97 commented on runtimes/peregrine/Cargo.toml:
```diff
@@ -125,6 +126,16 @@ frame-system-benchmarking           = { workspace = true, optional = true }
+pallet-ismp                = { workspace = true, features = ["unsigned"] }
+pallet-ismp-runtime-api    = { workspace = true }
+pallet-token-gateway       = { workspace = true }
+sp-mmr-primitives          = { workspace = true }
```
```suggestion
```

Ad96el replied:
[5fd3491](https://github.com/KILTprotocol/kilt-node/pull/888/commits/5fd3491e109b5ef2a210fb0e0ec6b7f008e62971)

Ad96el committed:
remove crate

Ad96el committed:
ismp

Ad96el committed:
use right dispatcher

Ad96el committed:
parachain-inherent

Ad96el committed:
enable indexing

Ad96el commented:
@Wizdave97 Thanks for the review 🙌

Much appreciated! I’ve addressed all your comments.
Indexing is enabled in [f57e137](https://github.com/KILTprotocol/kilt-node/pull/888/commits/f57e137b24fa7f2cf1ed7c578d4261fb17d2fc67), and I double-checked that the signedExtras match exactly.
I also updated the para_id of the test chain.

Wizdave97 commented:
Looks Good

Ad96el committed:
trait bounds

Ad96el committed:
trait bounds

Ad96el committed:
trait bounds

Ad96el committed:
make test pass
